src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsComplexTypeDefinition.cs

```diff
@@ -13,8 +13,9 @@ namespace Microsoft.OData.Edm.Csdl.CsdlSemantics
     /// <summary>
     /// Provides semantics for CsdlComplexType.
     /// </summary>
-    internal class CsdlSemanticsComplexTypeDefinition : CsdlSemanticsStructuredTypeDefinition, IEdmComplexType
+    internal class CsdlSemanticsComplexTypeDefinition : CsdlSemanticsStructuredTypeDefinition, IEdmComplexType, IEdmFullNamedElement

```

Comment:
>IEdmFullNamedElement [](start = 112, length = 20)

Rather than add to CsdlSemanticsComplexTypeDefinition and CsdlSemanticsEntityTypeDefinition, why don't we just add to CsdlSemanticsStructuredTypeDefinition?

Reply:
TBH, I didn't give much thought into this. Name property was defined on the derived type (CsdlSemanticsComplexTypeDefinition) and hence I added FullName there as well. If we need to clean up things then we should pick this as a separate work item to move properties that are defined on the derived class and move then to base. Agree?

Reply:
Agreed. Strange that Name is defined on the derived types, but makes sense to have Name and FullName co-located.

---
In reply to: [215790246](https://github.com/OData/odata.net/pull/1227#discussion_r215790246) [](ancestors = 215790246)